src/components/modal/mine-filter/mine-filter-nytt.tsx

```diff
@@ -1,37 +1,53 @@
+import {AppState} from '../../../reducer';
+import {Normaltekst} from 'nav-frontend-typografi';
+import {erTomtObjekt, feilValidering} from './mine-filter-utils';
+import {LagretFilterValideringsError} from './mine-filter-modal';
+import {ErrorModalType, MineFilterVarselModal} from './varsel-modal';
+import {lagreNyttFilter} from '../../../ducks/mine-filter';
+import {useRequestHandler} from '../../../hooks/use-request-handler';
+import {ListevisningType} from '../../../ducks/ui/listevisning';
+import {ThunkDispatch} from 'redux-thunk';
+import {AnyAction} from 'redux';
+import {SidebarTabInfo} from '../../../store/sidebar/sidebar-view-store';
 
-export function LagreNyttMineFilter(props: { filtergruppe: string, lukkModal }) {
-    const filterValg = useSelector((state: AppState) => props.filtergruppe === ListevisningType.minOversikt ? state.filtreringMinoversikt : state.filtreringEnhetensOversikt)
-    const data = useSelector((state: AppState) => state.mineFilter.data)
-    const [filterNavn, setFilterNavn] = useState("")
-    const [feilmelding, setFeilmelding] = useState({} as LagretFilterValideringsError)
-    const dispatch = useDispatch();
-    const requestHandler = useRequestHandler((state: AppState) => state.mineFilter.status, props.lukkModal)
+export function LagreNyttMineFilter(props: {filtergruppe: string; lukkModal}) {
+    const filterValg = useSelector((state: AppState) =>
+        props.filtergruppe === ListevisningType.minOversikt
+            ? state.filtreringMinoversikt
+            : state.filtreringEnhetensOversikt
+    );
+    const data = useSelector((state: AppState) => state.mineFilter.data);
+    const [filterNavn, setFilterNavn] = useState('');
+    const [feilmelding, setFeilmelding] = useState({} as LagretFilterValideringsError);
+
+    const dispatch: ThunkDispatch<AppState, any, AnyAction> = useDispatch();
+    const requestHandler = useRequestHandler((state: AppState) => state.mineFilter.status, props.lukkModal);
 
     const doLagreNyttFilter = (event) => {
-        event.preventDefault()
-        const feilValideringResponse = feilValidering(filterNavn, filterValg, data)
-        setFeilmelding(feilValideringResponse)
+        event.preventDefault();
+        const feilValideringResponse = feilValidering(filterNavn, filterValg, data);
+        setFeilmelding(feilValideringResponse);
 
         if (erTomtObjekt(feilValideringResponse)) {
-            requestHandler.setSaveRequestSent(true)
-            dispatch(lagreNyttFilter({
-                filterNavn: filterNavn,
-                filterValg: filterValg
-            }))
+            requestHandler.setSaveRequestSent(true);
+            dispatch(
+                lagreNyttFilter({
+                    filterNavn: filterNavn,
+                    filterValg: filterValg
+                })
+            ).then(() => {
+                dispatch({
+                    name: props.filtergruppe,
+                    type: 'sidebarTabEndret',
```

Comment:
maybe its better to use const SIDEBAR_TAB_ENDRET

Reply:
I attempted to implement what you described in a new commit (for both mine-filter-nytt.tsx
 and mine-filter-oppdater.tsx)

Reply:
Yep, its seems better now.